WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/InPersonPaymentsMenuViewModel.swift

```diff
@@ -87,32 +89,54 @@ final class InPersonPaymentsMenuViewModel {
     }
 
     private func checkShouldShowTapToPayFeedbackRow(siteID: Int64) {
-        let action = AppSettingsAction.loadFirstInPersonPaymentsTransactionDate(
-            siteID: siteID,
-            cardReaderType: .appleBuiltIn) { [weak self] firstTapToPayTransactionDate in
-                guard let self = self else { return }
-                guard let firstTapToPayTransactionDate = firstTapToPayTransactionDate,
-                      let thirtyDaysAgo = Calendar.current.date(byAdding: DateComponents(day: -30), to: Date()) else {
-                    return self.shouldShowTapToPayOnIPhoneFeedbackRow = false
-                }
+        Task { @MainActor in
```

Comment:
It would be nice to add tests for this logic

Reply:
I know, and I did add them initially, but because of the mix of Swift Concurrency and Store/Action dispatch, they don't work – it's something we've faced before.

I'm going to upgrade the screen to SwiftUI in a couple of weeks, and I'll try to pick this up along with that.